### PR TITLE
Add script to rebuild component refs

### PR DESCRIPTION
**Pull request by TravisEz13**
# PR Summary

Add script to rebuild WIX component refs

## PR Context

Someone (cc @SteveL-MSFT, @bergmeister), let the WIX file fragment be edited by hand and become corrupt.
This script examines the WXS fragment and looks for the corruption, fails the build, and outputs and correctly built WXS fragment.

Note,  **Please** do not hand edit the files.wxs file.  Always let the CI system, find the problems and fix them. 
If you try to hand edit, you risk us not being able to issues a security patch if needed.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

**Comment by TravisEz13** [comment]
This run produced the fixed wix file: https://dev.azure.com/powershell/PowerShell/_build/results?buildId=42371&view=artifacts&type=publishedArtifacts

File: tools/packaging/packaging.psm1 [hunk 1]
```diff
@@ -3044,6 +3044,35 @@ function Test-FileWxs
...
+        # add all the file components to the patch
+        foreach($component in $components)
+        {
+            $componentId = $component.Id
+            Write-Verbose "Removing $componentId" -Verbose
+            Remove-ComponentRefNode -Id $componentId -XmlDoc $newFilesAssetXml -XmlNsManager $xmlns
+        }
+
+        $componentGroups = @($newFilesAssetXml.GetElementsByTagName('ComponentGroup'))
+        $componentGroup = $componentGroups[0]
```
Comment (adityapatwardhan) [inline tools/packaging/packaging.psm1]: Add a comment why `[0]`

**Review by adityapatwardhan — APPROVED** [review]

**Comment by daxian-dbw** [comment]
@TravisEz13 Can you please update the PR description to include some context information about this change?

**Comment by bergmeister** [comment]
I am missing the context as well. Usually, I've used the auto-generated WiX file that `Start-PSPackage` produced when updates to DLLs were required due to a newer .Net Core version. Since the script does XML parsing to check the contents anyway, I don't understand what you mean by 'corrupted'

**Comment by TravisEz13** [comment]
@bergmeister is you used the autogenerated files, then you are not the cause.  Someone did not and the ComponentGroup section of the file did not contain all the updated components.

**Comment by TravisEz13** [comment]
This happened in the fork we used to update to .NET Core 3.0.  We squash merged that fork, so.... I cannot tell where exactly the issue occurred.

**Pull request merged**